File: DSA/SakshamDwivedi_590013147/Day19/D19P1.c
#include <stdio.h>

int arr[100];
int size;
int target;
int found = 0;

int main(){
    printf("Enter the size of the array (Less than 100) : ");
    scanf("%d", &size);
    printf("Enter the elements : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }
    printf("Enter the target value to be searched : ");
    scanf("%d", &target);
    int low = 0;
    int high = size - 1;
    int mid = 0;
    while (low <= high){
        mid = low + (high - low) / 2;
        if(arr[mid] == target){
            found = 1;
            break;
        }

        if(arr[mid] >= arr[low]){
            if(arr[low] <= target && target < arr[mid]){
                high = mid - 1;
            }
            else{
                low = mid + 1;
            }
        }
        else{
            if(arr[mid] < target && target <= arr[high]){
                low = mid + 1;
            }
            else{
                high = mid - 1;
            }
        }
    }

    if(found){
        printf("%d\n", mid);
    }
    else{
        printf("-1\n");
    }
}

File: DSA/SakshamDwivedi_590013147/Day19/D19P1.py
#Solution Function
def rotated_binary(arr, target):
    low = 0
    high = len(arr) - 1
    while low <= high:
        mid = low + (high - low) // 2       
        if(arr[mid] == target):
            return mid

        if arr[low] <= arr[mid]:
            if arr[low] <= target < arr[mid]:
                high = mid - 1
            else:
                low = mid + 1               
        else:
            if arr[mid] < target <= arr[high]:
                low = mid + 1
            else:
                high = mid - 1
                
    return -1

#Fixed Input
print(rotated_binary([4, 5, 6, 7, 0, 1, 2], 0))

#Custom Input
arr = list(int(x) for x in input("Enter the array : ").split())
target = int(input("Enter the target value to find : "))
print(rotated_binary(arr, target))
